app/controllers/posts_controller.rb

```diff
@@ -33,6 +34,18 @@ def create
     end
   end
 
+  def destroy
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+    if @post.destroy
+      Post.update_user_posts_counter(@user)
+      flash[:success] = 'Post was successfully deleted.'
+    else
+      flash[:error] = 'Something went wrong'
+    end
+    redirect_to user_posts_url
+  end
```

Comment:
- [ ] Nice work, but the destroy method is not working for the post when I try to delete, the update or delete on table "posts" violates foreign key constraint on table "likes"  is still referenced from table "likes".

Reply:
@dasileker that means you have to delete all the comments first then delete the post